sdk/identity/Azure.Identity/src/Credentials/AuthorizationCodeCredentialOptions.cs

```diff
@@ -26,7 +26,7 @@ public class AuthorizationCodeCredentialOptions : TokenCredentialOptions, ISuppo
         /// <summary>
         /// For multi-tenant applications, specifies additional tenants for which the credential may acquire tokens. Add the wildcard value "*" to allow the credential to acquire tokens for any tenant in which the application is installed.
         /// </summary>
-        public IList<string> AdditionallyAllowedTenants => AdditionallyAllowedTenantsCore;
+        public IList<string> AdditionallyAllowedTenants { get; internal set; } = new List<string>();
```

Comment:
I originally made this property readonly everywhere, and found that the impact to the tests would be quite large. Previously the tests were setting `AdditionallyAllowedTenantsCore` which is arguably much worse than this since not all options types used the base property in the same way. I think this is approach is much lower risk than that, and I found that other options types which expose readonly collection types in Azure.Core do this as well, such as `DiagnosticOptions.LoggedQueryParameters` and others.

If you feel strongly about this I can change it but it does end up impacting a large percentage of the tests.